### PR TITLE
rust/sip: register parser for tcp v10

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3351

Describe changes:
- Documentation updated
- SIP version parser fixed
- Probing parser functions removed

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=pr/1499
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

File: rust/src/sip/sip.rs [hunk 1]
```diff
@@ -518,20 +454,66 @@ pub unsafe extern "C" fn rs_sip_parse_response_tcp(
...
+        "NOTIFY\0",
+        "PUBLISH\0",
+        "MESSAGE\0",
+        "INFO\0",
+        "OPTIONS\0",
+    ];
+    let mut r = 0;
+    unsafe {
+        for method in methods {
+            let depth = (method.len() - 1) as u16;
```
Comment (catenacyber) [inline rust/src/sip/sip.rs:477]: you can still get rid of the `- 1` ;-)

File: rust/src/sip/parser.rs [hunk 2]
```diff
@@ -156,8 +160,10 @@ fn parse_request_uri(i: &[u8]) -> IResult<&[u8], &str> {
 }
 
 #[inline]
-fn parse_version(i: &[u8]) -> IResult<&[u8], &str> {
-    map_res(take_while1(is_version_char), std::str::from_utf8)(i)
+fn parse_version(i: &[u8]) -> IResult<&[u8], String> {
+    let (i, prefix) = map_res(tag("SIP/"), std::str::from_utf8)(i)?;
```
Comment (catenacyber) [inline rust/src/sip/parser.rs:164]: Do we really need the utf8 here ?

File: rust/src/sip/sip.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/sip/sip.rs:477]: Oh, seeing https://github.com/OISF/suricata/pull/9893#discussion_r1406676900

**Review by jufajardini — COMMENTED** [review]
Documentation changes look good to me :)

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on ASAN_TLPR1_suri.
  

Pipeline 16804

**Comment by catenacyber** [comment]
Yes but not sure we can share the pcap

**Comment by catenacyber** [comment]
OPTIONS is also a HTTP method...

**Comment by catenacyber** [comment]
> Please share the pcap file if possible, so I can confirm this.
Otherwise, feel free to debug it yourself if you are unable to share it (and if you want, obviously).

TL;DR let's remove the `OPTIONS` method for SIP recognition

If you want a pcap, you can try to create one by sending a `OPTIONS` request to a HTTP server (like `curl -X OPTIONS 127.0.0.1:8080/`

**Comment by glongo** [comment]
Replaced by https://github.com/OISF/suricata/pull/9967

**Comment by victorjulien** [comment]
So OPTIONS could be a nice candidate for a PM+PP setup both in HTTP and SIP, where the PP would do the additional validation of the protocol?
